### PR TITLE
CCD-4600-App-Insight Upgraded Java 17 and AppInsight v3 with some var…

### PR DESCRIPTION
**Pull request by emmanuel-francis95**
…iable changes

**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [ ] the `keep-helm` label has been added, if the helm release should be persisted after a successful build

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,11 +1,12 @@
 ARG JAVA_OPTS="-Djava.security.egd=file:/dev/./urandom"
-ARG APP_INSIGHTS_AGENT_VERSION=2.6.1
+ARG APP_INSIGHTS_AGENT_VERSION=3.4.9
```
Comment (timja) [inline Dockerfile:2]: should be added so renovate can update automatically:
```suggestion
# renovate: datasource=github-releases depName=microsoft/ApplicationInsights-Java
ARG APP_INSIGHTS_AGENT_VERSION=3.4.9
```

File: build.gradle [hunk 2]
```diff
@@ -171,12 +180,12 @@ dependencies {
     implementation group: 'org.apache.httpcomponents', name: 'httpclient', version: '4.5.13'
     implementation group: 'io.springfox', name: 'springfox-boot-starter', version: swagger2Version
 
-    compile "org.flywaydb:flyway-core:6.5.7"
-    compile group: 'org.yaml', name: 'snakeyaml', version: '1.32'
-    compile group: 'io.jsonwebtoken', name: 'jjwt', version:'0.9.1'
-    compile group: 'com.github.hmcts', name: 'service-auth-provider-java-client', version: '4.0.3'
-    compile group: 'com.github.hmcts', name: 'idam-java-client', version: '2.0.1'
-    compile group: 'org.bouncycastle', name: 'bcpkix-jdk15on', version: '1.70'
+    implementation "org.flywaydb:flyway-core:6.5.7"
```
Comment (timja) [inline build.gradle:183]: there's a lot more than just these to update

File: lib/applicationinsights.json [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
+  "connectionString": "${file:/mnt/secrets/ccd-data-store/app-insights-connection-string}",
+  "role": {
+    "name": "CCD-DataStore"
+  }
```
Comment (timja) [inline lib/applicationinsights.json:5]: could add health sampling at the same time:
https://github.com/hmcts/spring-boot-template/blob/74e6e173e44ee952d5b7609192d90c999db2ba61/lib/applicationinsights.json#L6-L22

File: lib/applicationinsights.json [hunk 4]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (timja) [inline lib/applicationinsights.json:1]: should remove AI-Agent.xml at the same time as it's no longer used:
https://github.com/hmcts/ccd-data-store-api/blob/master/lib/AI-Agent.xml

File: charts/ccd-data-store-api/values.preview.template.yaml [hunk 5]
```diff
@@ -27,8 +27,8 @@ java:
     DATA_STORE_DB_PASSWORD: "{{ .Values.postgresql.auth.password}}"
     DATA_STORE_DB_OPTIONS: "?stringtype=unspecified"
     DATA_STORE_DB_MAX_POOL_SIZE: 10
-    DEFINITION_STORE_HOST: http://ccd-definition-store-api-pr-575-java/
-    USER_PROFILE_HOST: http://ccd-user-profile-api-pr-399-java/
+    DEFINITION_STORE_HOST: http://ccd-definition-store-api-pr-1392-java/
+    USER_PROFILE_HOST: http://ccd-user-profile-api-pr-549-java/
```
Comment (RebeccaBaker) [inline charts/ccd-data-store-api/values.preview.template.yaml:31]: to be reverted upon merge

File: Jenkinsfile_CNP [hunk 6]
```diff
@@ -20,7 +20,7 @@ def branchesToSync = ['demo', 'ithc', 'perftest', 'develop']
 
 // Variables to switch pipeline logic and wiring per type of build
 
-def definitionStoreDevelopPr = "PR-575"  // This doesn't change frequently, but when it does, only change this value.
+def definitionStoreDevelopPr = "PR-1392"  // This doesn't change frequently, but when it does, only change this value.
```
Comment (RebeccaBaker) [inline Jenkinsfile_CNP:23]: to be reverted upon merge

File: charts/ccd-data-store-api/values.preview.template.yaml [hunk 7]
```diff
@@ -27,7 +27,7 @@ java:
     DATA_STORE_DB_PASSWORD: "{{ .Values.postgresql.auth.password}}"
     DATA_STORE_DB_OPTIONS: "?stringtype=unspecified"
     DATA_STORE_DB_MAX_POOL_SIZE: 10
-    DEFINITION_STORE_HOST: http://ccd-definition-store-api-pr-575-java/
+    DEFINITION_STORE_HOST: http://ccd-definition-store-api-pr-1392-java/
```
Comment (RebeccaBaker) [inline charts/ccd-data-store-api/values.preview.template.yaml:30]: to be reverted

File: charts/ccd-data-store-api/values.preview.template.yaml (hunk 5 above)
Comment (dinesh1patel) [reply charts/ccd-data-store-api/values.preview.template.yaml]: Reverted

**Review by dinesh1patel — APPROVED** [review]
LGTM

File: Jenkinsfile_CNP (hunk 6 above)
Comment (dinesh1patel) [reply Jenkinsfile_CNP]: Reverted

**Review by RebeccaBaker — APPROVED** [review]

**Pull request merged**